examples/tti/TTI_codegen.py

```diff
@@ -48,27 +51,95 @@ def damp_boundary(damp, nbp):
         # Initialize damp by calling the function that can precompute damping
         damp_boundary(self.damp.data, nbpml)
 
-        if len(self.damp.shape) == 2 and self.src.receiver_coords.shape[1] == 3:
-            self.src.receiver_coords = np.delete(self.src.receiver_coords, 1, 1)
-        if len(self.damp.shape) == 2 and self.data.receiver_coords.shape[1] == 3:
-            self.data.receiver_coords = np.delete(self.data.receiver_coords, 1, 1)
-
     def Forward(self, save=False, dse='advanced', auto_tuning=False,
-                cache_blocking=None, compiler=None, u_ini=None):
-        fw = ForwardOperator(self.model, self.src, self.damp, self.data,
+                cache_blocking=None, compiler=None, u_ini=None, legacy=True):
+        nt, nrec = self.data.shape
+        nsrc = self.source.shape[1]
+        ndim = len(self.damp.shape)
+        dt = self.dt
+        h = self.model.get_spacing()
+        dtype = self.damp.dtype
+        nbpml = self.model.nbpml
+
+        # uses space_order/2 for the first derivatives to
+        # have spc_order second derivatives for consistency
+        # with the acoustic kernel
+        u = TimeData(name="u", shape=self.model.get_shape_comp(),
+                     time_dim=nt, time_order=self.t_order,
+                     space_order=self.s_order/2,
+                     save=save, dtype=dtype)
+        v = TimeData(name="v", shape=self.model.get_shape_comp(),
+                     time_dim=nt, time_order=self.t_order,
+                     space_order=self.s_order/2,
+                     save=save, dtype=dtype)
+
+        u.pad_time = save
+        v.pad_time = save
+
+        if u_ini is not None:
+            u.data[0:3, :] = u_ini[:]
+            v.data[0:3, :] = u_ini[:]
+
+        # Create source symbol
+        p_src = Dimension('p_src', size=nsrc)
```

Comment:
One more thing: could in principle this be created internally directly by ``SourceLike`` ?